packages/core/src/tools/tool-compatibility/builder.test.ts

```diff
@@ -0,0 +1,273 @@
+
+    if (toolResult?.result?.success) {
+      return {
+        modelName: model.modelId,
+        modelProvider: model.provider,
+        testName: toolName,
+        status: 'success',
+        error: null,
+        receivedContext: toolResult.result.receivedContext,
+        testId,
+      };
+    } else {
+      const error = toolResult?.result?.error || response.text || 'Tool call failed or result missing';
+      return {
+        modelName: model.modelId,
+        testName: toolName,
+        modelProvider: model.provider,
+        status: 'failure',
+        error: error,
+        receivedContext: toolResult?.result?.receivedContext || null,
+        testId,
+      };
+    }
+  } catch (e: any) {
+    return {
+      modelName: model.modelId,
+      testName: toolName,
+      modelProvider: model.provider,
+      status: 'error',
+      error: e.message,
+      receivedContext: null,
+      testId,
+    };
+  }
+}
+
+describe('Tool Schema Compatibility', () => {
+  // Set a longer timeout for the entire test suite
+  const SUITE_TIMEOUT = 120000; // 2 minutes
+  const TEST_TIMEOUT = 60000; // 1 minute
+
+  if (!process.env.OPENROUTER_API_KEY) throw new Error('OPENROUTER_API_KEY environment variable is required');
+  const openrouter = createOpenRouter({ apiKey: process.env.OPENROUTER_API_KEY });
+
+  const modelsToTest = [
+    // Anthropic Models
+    openrouter('anthropic/claude-3.7-sonnet'),
+    openrouter('anthropic/claude-3.7-sonnet:thinking'),
+    openrouter('anthropic/claude-3.7-sonnet:beta'),
+    openrouter('anthropic/claude-3.5-sonnet'),
+    openrouter('anthropic/claude-3.5-haiku'),
+
+    // NOTE: Google models accept number constraints like numberLt, but the models don't respect it and returns a wrong response often
+    // Unions of objects are not supported
+    // // Google Models
+    openrouter('google/gemini-2.5-pro-preview-03-25'),
+    openrouter('google/gemini-2.5-flash-preview'),
+    openrouter('google/gemini-2.5-flash-preview:thinking'),
+    openrouter('google/gemini-2.0-flash-001'),
+    openrouter('google/gemini-2.0-flash-lite-001'),
+
+    // // OpenAI Models
+    openrouter('openai/gpt-4o-mini'),
+    openrouter('openai/gpt-4.1'),
+    openrouter('openai/gpt-4.1-mini'),
+    openrouter('openai/o4-mini-high'),
+    openrouter('openai/o3-mini'),
+    openrouter('openai/gpt-4o-2024-11-20'),
+    // openrouter disables structured outputs by default for o3-mini, so added in a reasoning model not through openrouter to test
+    openai('o3-mini'),
+    openai('o4-mini'),
+
+    // // Meta Models
+    // Meta seems to not handle tools correctly in general, so commenting out for now
+    // openrouter('meta-llama/llama-4-maverick'),
+
+    // // Other Models
+    // deepseek randomly doesn't call the tool so the check fails. It seems to handle the tool call correctly though when it does call it
+    // openrouter('deepseek/deepseek-chat-v3-0324'),
```

Comment:
same about temp

Reply:
deekseek seems to be more consistent if i remove temp and add a compat layer, but it still errors a good amount, so I won't add it to the test suite, but i'll leave the compat layer in

llama just seems to not care about temp or a compat layer. It just repeatedly does shit like this lol, not really anything we can do here.
```
Error for meta-llama/llama-4-maverick - stringCuid: Model tried to call unavailable tool 'tesTool_stringCuid'. Available tools: testTool_stringCuid.
```